### PR TITLE
SLS-478 Allow for testing of disaggregated block manager, deltas, and PALM without oligarch tables

### PR DESCRIPTION
**Pull request by korteland**

### CODE REVIEW

**Comment by korteland** [comment]
Writeup of the issues we found during testing is in a doc linked from the ticket.

File: test/suite/test_oligarch09.py [hunk 1]
```diff
@@ -66,12 +72,13 @@ def conn_extensions(self, extlist):
         DisaggConfigMixin.conn_extensions(self, extlist)
 
     def test_oligarch_read_write(self):
-        uri = "oligarch:test_oligarch09"
         create_session_config = 'key_format=S,value_format=S,block_compressor={}'.format(self.block_compress)
+        if self.uri.startswith("file"):
```
Comment (pmacko86) [inline test/suite/test_oligarch09.py:76]: Can you refactor this by adding a method that generates the config string for the corresponding URI (passed in as a parameter)?

**Review by pmacko86 — APPROVED** [review]
LGTM, with one suggestion.

File: test/suite/test_oligarch09.py (hunk 1 above)
Comment (korteland) [reply test/suite/test_oligarch09.py]: Good call! Thanks, done in 37dcc79.

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/btree/bt_handle.c [hunk 2]
```diff
@@ -540,8 +540,13 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt, bool is_ckpt)
     else
         F_CLR(btree, WT_BTREE_NO_CHECKPOINT);
 
-    /* Detect if the btree is disaggregated (for now, just use the file extension). */
-    if (WT_SUFFIX_MATCH(btree->dhandle->name, ".wt_stable")) {
+    /*
+     * Detect if the btree is disaggregated. TODO the file extension check should be replaced with
+     * something more robust.
+     */
```
Comment (ddanderson) [inline src/btree/bt_handle.c:546]: Using an oligarch table implies that the `.wt_stable` file must use disagg block manager right?
Maybe we should set `block_manager=disagg` in the config for the wt_stable when it is created in schema_create?  If that's in place, then "something more robust" in the comment, is merely removing the .wt_stable check, and just checking the `block_manager=` config here.

I think what you have here is fine - I'm just thinking out loud to the next step when the pantry block manager is removed.

**Review by ddanderson — APPROVED** [review]
LGTM - I made a comment, but it's mostly just thinking about next steps.

Thanks for doing this, it will be great to test the BM independently.

File: src/btree/bt_handle.c (hunk 2 above)
Comment (korteland) [reply src/btree/bt_handle.c:546]: Yes, that's a good call. I made SLS-527 to track this.

**Pull request merged**